python/gstaichi/lang/misc.py

```diff
@@ -303,6 +303,7 @@ def init(
     _test_mode: bool = False,
     enable_fallback: bool = True,
     require_version: str | None = None,
+    print_non_pure: bool = False,
```

Comment:
since it is a newly added function argument, it is helpful to also update the docstring below. e.g., what is `print_non_pure` for?